engine/src/main/java/org/hibernate/search/spi/impl/HashSetIndexedTypesSet.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.spi.impl;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.hibernate.search.spi.IndexedTypeIdentifier;
+import org.hibernate.search.spi.IndexedTypesSet;
+
+final class HashSetIndexedTypesSet implements IndexedTypesSet, Serializable {
```

Comment:
Same remark as in `ConcurrentIndexedTypeMap`:

Maybe we'd better have an implementation simply accept a java.util.function.Supplier<? extends Set<IndexedTypeIdentifier>> in its main constructor? Then you'd only have to write something like new IndexedTypesSet<>(HashSet::new). We'd avoid duplicating the whole inheritance tree of java.util.Map...